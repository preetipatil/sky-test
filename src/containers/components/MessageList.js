import React from 'react';
import MessageItem from './MessageItem';
import PropTypes from 'prop-types';
import { findUserDetails, sortByDate } from '../utils';

function NoMessages() {
  return <div>No messages found</div>;
}

class MessageList extends React.Component {
  renderMessageList({ messages, members }) {
    const posts = sortByDate(messages);
    const findUser = findUserDetails.bind(this, members || []);
    return (
      <ul>
        {posts.map(post => {
          let user = findUser(post.userId);
          return (
            <MessageItem
              key={post.id}
              chatMessage={post.message}
              email={user.email}
              avatar={user.avatar}
              timestamp={post.timestamp}
            />
          );
        })}
      </ul>
    );
  }

  render() {
    return (
      <div>
        <h1>LOADED</h1>
        {this.props.messages ? this.renderMessageList(this.props) : <NoMessages />}
      </div>
    );
  }
}

MessageList.propTypes = {
  messages: PropTypes.array,
  members: PropTypes.array,
};
export default MessageList;
