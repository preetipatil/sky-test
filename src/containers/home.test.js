import React from 'react';
import { shallow } from 'enzyme';

import Home, { Loading, Error } from './home';
import MessageList from './components/MessageList';

describe('Home Page Test', () => {
  it('should renders messageList correctly', () => {
    const props = {
      messages: { loading: false, messages: [], error: null },
      members: { loading: false, members: [], error: null },
      loadMessages: () => {},
      loadMembers: () => {},
    };

    const tree = shallow(<Home.WrappedComponent {...props} />);
    expect(tree.containsAnyMatchingElements([<MessageList />])).toEqual(true);
    //expect(tree).toMatchSnapshot();
  });

  it('should match messageList snapshot correctly', () => {
    const props = {
      messages: { loading: false, messages: [], error: null },
      members: { loading: false, members: [], error: null },
      loadMessages: () => {},
      loadMembers: () => {},
    };

    const tree = shallow(<Home.WrappedComponent {...props} />);
    expect(tree.containsAnyMatchingElements([<MessageList />])).toEqual(true);
    expect(tree).toMatchSnapshot();
  });

  it('should show LOADING when Api is loading', () => {
    const props = {
      messages: { loading: true, messages: [], error: null },
      members: { loading: true, members: [], error: null },
      loadMessages: () => {},
      loadMembers: () => {},
    };

    const wrapper = shallow(<Home.WrappedComponent {...props} />);
    expect(wrapper.contains(<Loading />)).toEqual(true);
    //expect(wrapper).toMatchSnapshot();
  });

  it('should show ERROR IN LOADING DATA when Api is failed to load', () => {
    const props = {
      messages: { loading: false, messages: [], error: 'Failed' },
      members: { loading: false, members: [], error: null },
      loadMessages: () => {},
      loadMembers: () => {},
    };

    const tree = shallow(<Home.WrappedComponent {...props} />);
    expect(tree.contains(<Error />)).toEqual(true);
    //expect(wrapper).toMatchSnapshot();
  });
});
