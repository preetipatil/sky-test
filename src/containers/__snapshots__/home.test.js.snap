// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Test should match messageList snapshot correctly 1`] = `
<MessageList
  members={Array []}
  messages={Array []}
/>
`;

exports[`Home Page Test should renders correctly 1`] = `
<MessageList
  members={Array []}
  messages={Array []}
/>
`;

exports[`Home Page Test should renders messageList correctly 1`] = `
<MessageList
  members={Array []}
  messages={Array []}
/>
`;
